### PR TITLE
Fix battery sim

### PR DESCRIPTION
**Pull request by Daniel1464**
Currently, MapleSim's battery sim can return NaN in fringe scenarios, causing RobotController.getBatteryVoltage() to return invalid values. This can be detrimental if PathPlanner's SwerveSetpointGenerator is used(or other calls to RobotController.getBatteryVoltage are used in the code), as they rely on RobotController.getBatteryVoltage.

This commit gives two safeguard:
1. It halves the current of the battery sim
2. It handles NaN scenarios by printing an error msg and defaulting to 12 volts instead

Currently, i dont exactly know what causes this.

### CODE REVIEW

**Comment by Daniel1464** [comment]
Note: defaulting to the brownout voltage gives like 6.75 volts to it, which ends up slowing down pathplanner's setpoint gen a lot, so i decided to do 12 volts instead

File: project/src/main/java/org/ironmaple/simulation/motorsims/SimulatedBattery.java [hunk 1]
```diff
@@ -28,38 +31,42 @@ public static void addMotor(MapleMotorSim mapleMotorSim) {
     }
 
     public static void flush() {
-        final double totalCurrentAmps = electricalAppliances.stream()
-                .mapToDouble(currentSupplier -> currentSupplier.get().in(Amps))
-                .sum()
+        double totalCurrentAmps = electricalAppliances.stream()
+                        .mapToDouble(currentSupplier -> currentSupplier.get().in(Amps))
+                        .sum()
                 / 2.0;
```
Comment (mmilunicmobile) [inline project/src/main/java/org/ironmaple/simulation/motorsims/SimulatedBattery.java:37]: Shouldn't this not be divided by 2? I believe that was a temporary fix for current drawing too much, but we figured out the underlying cause of that.

**Review by mmilunicmobile — CHANGES_REQUESTED** [review]
We shouldn't still be dividing the total current by 2.

File: project/src/main/java/org/ironmaple/simulation/motorsims/SimulatedBattery.java (hunk 1 above)
Comment (catr1xLiu) [reply project/src/main/java/org/ironmaple/simulation/motorsims/SimulatedBattery.java]: I guess you're right.  I was also wondering why it was there.

**Comment by catr1xLiu** [comment]
I added the Javadocs and I think this is ready for merging.

@Daniel1464 @mmilunicmobile Thank you so much for working on this, the battery simulation is working great now!

**Pull request merged**